uber/model_checks.py

```diff
@@ -264,24 +258,41 @@ def dealer_needs_group(attendee):
 
 @validation.Attendee
 def dupe_badge_num(attendee):
-    if c.NUMBERED_BADGES and attendee.badge_num != 0 and attendee.session.query(Attendee)\
-            .filter(Attendee.id != attendee.id)\
-            .filter_by(badge_type=attendee.badge_type, badge_num=attendee.badge_num).count():
-        return 'Another attendee already exists with that badge number!'
+    if (attendee.badge_num != attendee.orig_value_of('badge_num') or attendee.is_new)\
+            and c.NUMBERED_BADGES and attendee.badge_num and not c.SHIFT_CUSTOM_BADGES:
+        with Session() as session:
+            existing = session.query(Attendee)\
+                .filter_by(badge_type=attendee.badge_type, badge_num=attendee.badge_num)
+            if existing.count():
+                return 'That badge number already belongs to {!r}'.format(existing.first().full_name)
 
 
 @validation.Attendee
 def invalid_badge_num(attendee):
-    if c.NUMBERED_BADGES:
+    if c.NUMBERED_BADGES and attendee.badge_num:
         try:
```

Comment:
quick thing: I'm on a small crusade to as we go replace any exception handling code with more specific exceptions if possible, mostly because the exceptions are now logged and if we encounter an unexpected exception it's good to see it.

Since we:
1) already check for None before this
2) this try/except is trying to make sure that badge_num is an int
3) casting something not an int with int() results in a ```ValueError```

could you change this up so that we catch only ```ValueError``` instead of all exceptions?

then, bonus, if you felt like it you could write a unit test to catch this case by making sure it throws an ValueError exception after intentionally monkeypatching attendee.badge_num with a string.